Kmesh panic after applying a ServiceEntry

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:

Running in ads mode,

After creating a SE with this guide, 
https://istio.io/latest/docs/tasks/traffic-management/egress/egress-control/#access-an-external-http-service

I found that kmesh panicked. 


**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

stack traces
```
time="2024-04-22T07:27:58Z" level=info msg="controller Start successful" subsys=manager
time="2024-04-22T07:27:58Z" level=info msg="dump StartServer successful" subsys=manager
time="2024-04-22T07:27:58Z" level=info msg="start write CNI config\n" subsys="cni installer"
time="2024-04-22T07:27:58Z" level=info msg="kmesh cni use chained\n" subsys="cni installer"
panic: runtime error: index out of range [3] with length 0

goroutine 16 [running]:
encoding/binary.littleEndian.Uint32(...)
	/usr/local/go/src/encoding/binary/binary.go:84
kmesh.net/kmesh/pkg/nets.ConvertIpToUint32({0xc0009bd2d0?, 0x0?})
	/kmesh/pkg/nets/nets.go:34 +0x68
kmesh.net/kmesh/pkg/controller/envoy.newApiSocketAddress(0xc000085aa8?)
	/kmesh/pkg/controller/envoy/ads_loader.go:144 +0xa7
kmesh.net/kmesh/pkg/controller/envoy.newApiClusterLoadAssignment(0xc000d55500)
	/kmesh/pkg/controller/envoy/ads_loader.go:107 +0x28d
kmesh.net/kmesh/pkg/controller/envoy.(*AdsLoader).CreateApiClusterByCds(0xc000852a50, 0x2, 0xc000798a00)
	/kmesh/pkg/controller/envoy/ads_loader.go:71 +0x150
kmesh.net/kmesh/pkg/controller/envoy.(*ServiceEvent).handleCdsResponse(0xc000859320, 0xc0008c4080)
	/kmesh/pkg/controller/envoy/ads_event.go:168 +0x532
kmesh.net/kmesh/pkg/controller/envoy.(*ServiceEvent).processAdsResponse(0xc000859320, 0xc0008c4080)
	/kmesh/pkg/controller/envoy/ads_event.go:126 +0x174
kmesh.net/kmesh/pkg/controller/envoy.(*AdsStream).AdsStreamProcess(0xc00078e6a8)
	/kmesh/pkg/controller/envoy/ads_stream.go:63 +0x7f
kmesh.net/kmesh/pkg/controller.(*XdsClient).clientResponseProcess(0xc0008be370, {0x32569a0, 0xc0008be3c0})
	/kmesh/pkg/controller/client.go:128 +0x125
created by kmesh.net/kmesh/pkg/controller.(*XdsClient).Run in goroutine 1
	/kmesh/pkg/controller/client.go:151 +0xd1

```

--- a/pkg/controller/envoy/ads_loader.go
+++ b/pkg/controller/envoy/ads_loader.go
@@ -106,7 +106,8 @@ func newApiClusterLoadAssignment(
 			apiEndpoint := &endpoint_v2.Endpoint{
 				Address: newApiSocketAddress(endpoint.GetEndpoint().GetAddress()),
 			}
-			if apiEndpoint.GetAddress() == nil {
+			if apiEndpoint.GetAddress() == nil || apiEndpoint.Address.Ipv4 == 0 {
+				log.Warnf("cluster %s endpoint address %v is nil or invalid", loadAssignment.GetClusterName(), endpoint.GetEndpoint().GetAddress())
 				continue
 			}
 			apiLocalityLb.LbEndpoints = append(apiLocalityLb.LbEndpoints, apiEndpoint)

--- a/pkg/nets/nets.go
+++ b/pkg/nets/nets.go
@@ -12,9 +12,6 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
-
- * Author: LemmyHuang
- * Create: 2021-12-07
  */
 
 // Package nets : net connection provider

--- a/pkg/nets/nets.go
+++ b/pkg/nets/nets.go
@@ -28,10 +28,17 @@ import (
 // ConvertIpToUint32 converts ip to little-endian uint32 format
 func ConvertIpToUint32(ip string) uint32 {
 	netIP := net.ParseIP(ip) // BigEndian
+	if netIP == nil {
+		return 0
+	}
+	// TODO: is this right?
 	if len(netIP) == net.IPv6len {
 		return binary.LittleEndian.Uint32(netIP.To4())
 	}
-	return binary.LittleEndian.Uint32(netIP)
+	if len(netIP) == net.IPv4len {
+		return binary.LittleEndian.Uint32(netIP)
+	}
+	return 0
 }
 
 // ConvertUint32ToIp converts big-endian uint32 to ip format

--- a/pkg/nets/nets_test.go
+++ b/pkg/nets/nets_test.go
@@ -1,0 +1,33 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nets
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_ConvertIpToUint32(t *testing.T) {
+	ip := "192.168.0.1"
+	val := ConvertIpToUint32(ip)
+	assert.Equal(t, 0x100a8c0, val)
+
+	// It can not panic even for invalid ip
+	val = ConvertIpToUint32("a.b.c.d")
+	assert.Equal(t, uint32(0), val)
+}

--- a/pkg/nets/nets_test.go
+++ b/pkg/nets/nets_test.go
@@ -25,7 +25,7 @@ import (
 func Test_ConvertIpToUint32(t *testing.T) {
 	ip := "192.168.0.1"
 	val := ConvertIpToUint32(ip)
-	assert.Equal(t, 0x100a8c0, val)
+	assert.Equal(t, uint32(0x100a8c0), val)
 
 	// It can not panic even for invalid ip
 	val = ConvertIpToUint32("a.b.c.d")

